client/checker_client/cli.py

```diff
@@ -0,0 +1,229 @@
+import click
+
+from checker_client import checker as checker_lib
+import pytezos
+from pathlib import Path
+from dataclasses import dataclass
+from marshmallow import Schema, fields
+from marshmallow.decorators import post_load
+import portpicker
+import json
+
+CONFIG_FILE = Path.home().joinpath(Path(".checker"))
+
+
+@dataclass
+class Config:
+    path: str
+    sandbox_port: int = 18731
+    sandbox_container: str = "checker-sandbox-node"
+    tezos_address: str = "http://127.0.0.1"
+    tezos_port: int = 18731
+    tezos_key: str = "edsk3RFfvaFaxbHx8BMtEW1rKQcPtDML3LXjNqMNLCzC3wLC1bWbAt"  # bob's private key from "edobox info"
+    ctez_address: str = ""
+    oracle_address: str = ""
+    checker_address: str = ""
+
+    @staticmethod
+    def load(file: Path):
+        with file.open("r") as f:
+            return ConfigSchema().loads(f.read())
+
+    def dump(self):
+        with open(self.path, "w") as f:
+            f.write(ConfigSchema().dumps(self))
+
+
+class ConfigSchema(Schema):
+    path = fields.String(required=True)
+    sandbox_container = fields.String()
+    sandbox_port = fields.Int()
+    sandbox_container = fields.String()
+    tezos_address = fields.String()
+    tezos_port = fields.Int()
+    tezos_key = fields.String()
+    ctez_address = fields.String()
+    oracle_address = fields.String()
+    checker_address = fields.String()
+
+    @post_load
+    def make(self, data, **kwargs):
+        return Config(**data)
+
+
+@click.group()
+@click.pass_context
+@click.option(
+    "--config",
+    "config_file",
+    default=CONFIG_FILE,
+    help="path to CLI config file",
+    show_default=True,
+)
+def cli(ctx, config_file):
+    """Checker command line utilities
+
+    By default this tool caches options for things like addresses in the
+    specified config file to make chaining calls easier.
```

Comment:
To me, the word "caches" was a bit misleading since it implies a performance optimisation. I would prefer something like "remembers".

Also, we might want to explain what exactly is going on a bit more; something like "the results of a command becomes the default arguments of the following commands" (with a nicer wording).